modules/ml/src/main/java/org/apache/ignite/ml/dataset/UpstreamTransformer.java

```diff
@@ -28,15 +27,8 @@
  * @param <V> Type of values in the upstream.
  */
 @FunctionalInterface
-public interface UpstreamTransformer<K, V> extends Serializable {
-    /**
-     * Perform transformation of upstream.
-     *
-     * @param rnd Random numbers generator.
-     * @param upstream Upstream.
-     * @return Transformed upstream.
-     */
-    // TODO: IGNITE-10296: Inject capabilities of randomization through learning environment.
-    // TODO: IGNITE-10297: Investigate possibility of API change.
-    public Stream<UpstreamEntry<K, V>> transform(Random rnd, Stream<UpstreamEntry<K, V>> upstream);
+public interface UpstreamTransformer<K, V> extends IgniteFunction<Stream<UpstreamEntry<K, V>>, Stream<UpstreamEntry<K, V>>> {
```

Comment:
Do you think it's a good idea to inherit from `IgniteFunction`? We're loosing clear `transform` signature, but I don't see any advantages.

Reply:
I guess, you're right, fixed.